src/python/espressomd/utils.pyx

```diff
@@ -141,42 +141,110 @@ cdef check_range_or_except(D, name, v_min, incl_min, v_max, incl_max):
             raise ValueError("In " + name + ": Value " + str(x) + " is out of range " + ("[" if incl_min else "]") +
                              str(v_min) + "," + str(v_max) + ("]" if incl_max else "["))
 
+
 def to_char_pointer(s):
     """
     Returns a char pointer which contains the information of the provided python string.
-    
+
     Parameters
     ----------
     s : :obj:`str`
 
     """
     if isinstance(s, unicode):
-        s = (<unicode>s).encode('utf8')
+        s = ( < unicode > s).encode('utf8')
     return s
 
+
 def to_str(s):
     """
     Returns a python string.
-    
+
     Parameters
     ----------
     s : char*
 
     """
     if type(s) is unicode:
-        return <unicode>s
+        return < unicode > s
     elif PY_MAJOR_VERSION >= 3 and isinstance(s, bytes):
-        return (<bytes>s).decode('ascii')
+        return ( < bytes > s).decode('ascii')
     elif isinstance(s, unicode):
         return unicode(s)
     else:
         return s
 
 
+class array_locked(np.ndarray):
+    """
+    Returns a non-writable numpy.ndarray with a special error message upon usage
+    of __setitem__  or in-place operators. Cast return in __get__ of array
+    properties to array_locked to prevent these operations. 
+
+    """
+
+    ERR_MSG = "Array properties return non-writable arrays \
+    and can only be modified as a whole, not in-place or componentwise. \
+    Use numpy.copy(<ESPResSo array property>) to get a writable copy."
+
+    def __new__(cls, input_array):
+        obj = np.asarray(input_array).view(cls)
+        obj.flags.writeable = False
+        return obj
+
+    def __repr__(self):
+        return repr(np.array(self))
+
+    def __array_finalize__(self, obj):
+        if obj is None:
+            return
+
+    def __setitem__(self, i, val):
+        raise UserWarning(array_locked.ERR_MSG)
```

Comment:
Should be ValueError, not UserError. Numpy raises `ValueError("assignment destination is read-only")`.